core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -445,8 +445,9 @@ class ProducerStateManager(val topicPartition: TopicPartition,
             lastMapOffset = lastSnapOffset
             return
           } catch {
-            case e: CorruptSnapshotException =>
-              error(s"Snapshot file at ${file.getPath} is corrupt: ${e.getMessage}")
+            case e: Exception =>
```

Comment:
Btw, I also agree that `CorruptSnapshotException` would be more appropriate if the version is wrong.